src/python/grpcio/grpc/experimental/aio/_call.py

```diff
@@ -182,11 +183,23 @@ def done(self) -> bool:
         """
         return self._state is not _RpcState.ONGOING
 
-    async def initial_metadata(self):
-        raise NotImplementedError()
+    async def initial_metadata(self) -> Sequence[Tuple[Text, AnyStr]]:
```

Comment:
Returning a plain old `Dict` would lose the order of the metadata entries, which we currently preserve with `Sequence[Tuple[Text, AnyStr]]`. Perhaps [`OrderedDict`](https://docs.python.org/2/library/collections.html#ordereddict-objects) instead?

Reply:
It's a pity that we can not rely on default dict implementation that
preserves insertion order since Python 3.6, because we are giving support
to python 3.5+.

In any case, agree on using ordered dict.